spec/services/replication_transfer_updater_spec.rb

```diff
@@ -0,0 +1,279 @@
+
+require 'rails_helper'
+
+describe ReplicationTransferUpdater do
+
+  describe "::build_update" do
+    let(:record) { Fabricate(:replication_transfer) }
+    it "returns a subclass of ReplicationTransferUpdate" do
+      allow(ReplicationTransferUpdater::CancelUpdate).to receive(:matching_update?).and_return true
+      expect(described_class.build_update(record,{}).class).to eql(ReplicationTransferUpdater::CancelUpdate)
+    end
+    it "returns a generic ReplicationTransferUpdate when no specific matches exist" do
+      expect(described_class.build_update(record,{}).class).to eql(ReplicationTransferUpdater::ReplicationTransferUpdate)
+    end
+  end
+
+
+  describe ReplicationTransferUpdater::ReplicationTransferUpdate do
+    describe "#we_requested?" do
+      let(:our_transfer) { Fabricate.build(:replication_transfer, from_node: Fabricate.build(:local_node)) }
+      let(:their_transfer) { Fabricate.build(:replication_transfer) }
+      it "is true if record.from_node is the local node" do
+        expect(described_class.new(our_transfer, nil).we_requested?).to be true
+      end
+      it "is false if it was created by another" do
+        expect(described_class.new(their_transfer, nil).we_requested?).to be false
+      end
+    end
+  end
+
+
+  let(:updated) {
+    updated = ReplicationTransferAdapter.from_model(record).to_params_hash
+    updated.merge!(changes)
+  }
+
+  describe ReplicationTransferUpdater::CancelUpdate do
+    let(:update) { described_class.new(record, updated) }
+    let(:record) { Fabricate(:replication_transfer) }
+    let(:changes) { {cancelled: true, cancel_reason: "test"} }
+    describe "::matching_update?" do
+      it "matches old.cancelled == false, new.cancelled == true" do
+        expect(described_class.matching_update?(record, updated)).to be true
+      end
+    end
+    describe "#update" do
+      it "cancels the record" do
+        update.update
+        expect(record.cancelled).to be true
+      end
+      it "sets the cancel reason" do
+        update.update
+        expect(record.cancel_reason).to eql("test")
+      end
+      it "returns true on success" do
+        allow(record).to receive(:cancel!).and_return true
+        expect(update.update).to be true
+      end
+      it "returns false on failure" do
+        allow(record).to receive(:cancel!).and_return false
+        expect(update.update).to be false
+      end
+    end
+  end
+
+
+  let(:fixity_alg) { Fabricate(:fixity_alg) }
+  let(:bag) {
+    bag = Fabricate(:bag_without_digests)
+    bag.message_digests << Fabricate.build(:message_digest, fixity_alg: fixity_alg, value: correct_fixity)
+    bag.save!
+    bag
+  }
+  let(:transfer) { Fabricate(:replication_transfer,
+    fixity_alg: fixity_alg,
+    bag: bag,
+    from_node: Fabricate(:local_node))
+  }
+
+  describe ReplicationTransferUpdater::FixityValueUpdate do
+    let(:update) { described_class.new(record, updated) }
+    let(:fixity_alg) { Fabricate(:fixity_alg) }
```

Comment:
This `fixity_alg` duplicates the same thing in the scope above, so it should be OK to remove it in this scope.